Tests/tinys3Tests/model/MultipartUploadFileTests.swift

```diff
@@ -48,13 +48,25 @@ final class MultipartUploadFileTests: XCTestCase {
     }
 
     func testThatPartSizeCalculatorReturnsAppropriateSizes() {
-        XCTAssertEqual(PartSizeCalculator.calculate(basedOn: 4094), 4094)                     //    4kb
-        XCTAssertEqual(PartSizeCalculator.calculate(basedOn: 4_094_000), 4_094_000)           //    4mb
-        XCTAssertEqual(PartSizeCalculator.calculate(basedOn: 40_094_000), 5_000_000)          //   40mb
-        XCTAssertEqual(PartSizeCalculator.calculate(basedOn: 400_094_000), 8_001_880)         //  400mb
-        XCTAssertEqual(PartSizeCalculator.calculate(basedOn: 4_000_094_000), 80_001_880)      //    4gb
-        XCTAssertEqual(PartSizeCalculator.calculate(basedOn: 40_000_094_000), 800_001_880)    //   40gb
-        XCTAssertEqual(PartSizeCalculator.calculate(basedOn: 400_000_094_000), 4_900_000_000) //  400gb
-        XCTAssertEqual(PartSizeCalculator.calculate(basedOn: 400_000_094_000), 4_900_000_000) //    4tb
+        // swiftlint:disable colon comma
+        XCTAssertEqual(PartSizeCalculator.calculate(basedOn:   4.KB),   4    .KB) //   4KB
+        XCTAssertEqual(PartSizeCalculator.calculate(basedOn:   4.MB),   4    .MB) //   4MB
```

Comment:
Interesting issue (that this PR doesn't necessarily need to address) – these file sizes are invalid – they're below the minimum allowed file size for a multipart upload part.

There are a few different ways we could handle this, but _should we_?

Reply:
No this is valid.

If the total file size is 4MB then the multipart upload will consist of **a single part**, which will be 4MB.

The part size limit does not apply to the **last** part of a multipart upload (which makes sense because the last part is always "the remainder"/the leftovers). So in the case of a file <5MB in total the single part will also be the last part and is allowed to be <5MB.

Reply:
https://docs.aws.amazon.com/AmazonS3/latest/userguide/qfacts.html

> There is no minimum size limit on the last part of your multipart upload.

Reply:
FYI: I've opened https://github.com/Automattic/hostmgr/pull/94 as a follow-up to address an off-by-one error that I noticed on the implementations of how those `partSize` values are used to slice the `Range` objects and read the `FileHandle` (TL;DR: we were always sending one byte more than the computed `partSize` for each part)

Reply:
Can you review https://github.com/Automattic/hostmgr/pull/72? 

My understanding is that if there's only a single part that's under 5MB we can't use multipart at all, but maybe I'm wrong

Reply:
Oh interesting! ok 👍
That wasn't my original understanding of the docs (and I guess the initial implementation of `PartSizeCalculator` that contained that `min(fileSize, 5MB)` also biased my understanding after all)

That means we'll have to implement non-Multipart upload in those cases indeed, which I'll do on a separate PR (on top of **both** #92 and #94). Or at least I'll create an issue for it on my tomorrow to note to tackle it later 👍

Reply:
- Tracked the issue for <5MB files in https://github.com/Automattic/hostmgr/issues/95
 - Updated code `/* comments */` in https://github.com/Automattic/hostmgr/pull/94 to adjust to our new understanding.